Modules/mathmodule.c

```diff
@@ -2449,14 +2457,37 @@ vector_norm(Py_ssize_t n, double *vec, double max, int found_nan)
     if (max == 0.0 || n <= 1) {
         return max;
     }
+    frexp(max, &max_e);
+    if (max_e >= -1023) {
```

Comment:
Do we need scaling at all for small absolute `max_e`? E.g. for `-1024/2 < max_e < (1024-64)/2`. Could not it speed up the loop in most common cases?

Reply:
Several thoughts:
* Yes, I believe it is needed. 
* The speed effect is minimal. 
* There is no common case for values this small. This is rare.
* Removing the scaling would be a step backwards.

Reply:
I think that the opposite case, values larger than 1e150 or smaller than 1e-150 (except exact 0) are rare.

Reply:
The scaling isn't just for overflow and underflow.  It is a pre-condition for the accurate summation logic. 

How about we get this code in.  Later you can attempt to further optimize it without losing accuracy, commutativity, and monotonicity.

We've got a straight win here — it's more accurate and faster than what we have now.  I'm happy with that.

Reply:
I updated the comment to indicate why scaling or division by *max* is always desirable, even when not needed for overflow/underflow protection.